engine-core/src/main/java/io/nosqlbench/engine/core/lifecycle/ActivityExecutor.java

```diff
@@ -248,17 +247,14 @@ public boolean finishAndShutdownExecutor(int secondsToWait) {
         return wasStopped;
     }
 
-
     /**
      * Listens for changes to parameter maps, maps them to the activity instance, and notifies all eligible listeners of
      * changes.
      */
     @Override
     public synchronized void handleParameterMapUpdate(ParameterMap parameterMap) {
 
-        if (activity instanceof ActivityDefObserver) {
```

Comment:
always true due to type hierarchy... so it was basically a null check but activiy is used below as if it can never be null